util/activity-report/avg-tx-cost.js

```diff
@@ -45,71 +42,14 @@ async function dbQueryAvgTxCost(blocks, chainTableName) {
     ON t.block_id = lb.id
     WHERE t.fees_paid != 0  
   `;
-  const query = format(queryStr, chainTableName, blocks, chainTableName);
+  const tableName = chainTableNames[chain];
+  const query = format(queryStr, tableName, blocks, tableName);
   const [avgTxCost] = (await db.query(query)).rows;
   if (!avgTxCost?.avg_tx_cost_usd) throw new Error('DB records not found');
-  return {
-    time: new Date(),
-    ...avgTxCost,
-  };
-}
-
-function getFromDbOrLongStorage(blocks, chainTableName) {
-  return new Promise((resolve) => {
-    const getCached = () =>
-      resolve(cache.get(getLongStorageId(chainTableName)));
-    // try to get db response before the deadline
-    (async () => {
-      try {
-        const avgTxCost = await dbQueryAvgTxCost(blocks, chainTableName);
-        // each time we are able to successfully get the query result
-        // store these values + the timestamp in memory
-        cache.set(getLongStorageId(chainTableName), avgTxCost);
-        resolve(avgTxCost);
-      } catch (error) {
-        // each time we are unsuccessful in making the query (e.g. DB went down)
-        // retrieve the values plus timestamp from the long storage
-        getCached();
-      }
-    })();
-    // if waiting longer than deadline, read from the long storage
-    setTimeout(getCached, databaseDelayLimit * 1000);
-  });
-}
-
-function verifyParams(blocks) {
-  const blocksRange = {
-    lower: 1,
-    upper: 1000,
-  };
-  if (!(blocks >= blocksRange.lower && blocks <= blocksRange.upper))
-    throw new Error(
-      `Number of blocks should be within range ${blocksRange.lower} to ${blocksRange.upper}, specified value was: ${blocks}`,
-    );
-}
-
-async function queryAvgTxCost({ blocks = 100, chain = 'rsk_mainnet' }) {
-  verifyParams(blocks);
-  const chainTableName = getChainTableName(chain);
-  // Try to read from short term storage.
-  // Less than 10 minutes after last DB query
-  // --> return cached value, no query performed
-  let avgTxCost = cache.get(getShortStorageId(chainTableName));
-  // When the short term storage ttl is expired:
-  // 10 minutes or longer --> perform the query, and return new value
-  if (!avgTxCost) {
-    // read from either DB or long storage (whichever available)
-    avgTxCost = await getFromDbOrLongStorage(blocks, chainTableName);
-    // write received data to short storage
-    cache.set(getShortStorageId(chainTableName), avgTxCost, shortStorageTtl);
-  }
-  return {
-    blocks,
-    chain,
-    ...avgTxCost,
-  };
+  return avgTxCost;
 }
 
 module.exports = {
-  queryAvgTxCost,
+  queryDb,
+  verifyBlocks,
 };
```

Comment:
I put here all functions related to this particular endpoint. It's easy to import them from here to the config file